### PR TITLE
PIN-4159 - Added tenant distribution metric

### PR DESCRIPTION
**Pull request by Carminepo2**

### CODE REVIEW

File: jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    )
+    .toArray()
+
+  const [onBoardedTenantIds, activeAgreements] = await Promise.all([
+    onBoardedTenantIdsQueryPromise,
+    activeAgreementsQueryPromise,
+  ])
+
+  const agreementsConsumersIds = uniq(activeAgreements.map((agreement) => agreement.consumerId))
+  const agreementsProducersIds = uniq(activeAgreements.map((agreement) => agreement.producerId))
```
Comment (galales) [inline jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts]: may be worth to use [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#performance) in this case?

File: jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  function resolveTenantDistribution(tenantId: string): void {
+    const isTenantProducer = checkIsProducer(tenantId)
+    const isTenantConsumer = checkIsConsumer(tenantId)
+
+    if (isTenantProducer && isTenantConsumer) bothConsumersAndProducers.count++
+    else if (isTenantProducer) onlyProducers.count++
+    else if (isTenantConsumer) onlyConsumers.count++
+    else onlyFirstAccess.count++
```
Comment (galales) [inline jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts:78]: you probably already know it, but now the presence of the selfcareId does not mean that the Institutions has actually performed the first login

**Review by galales — APPROVED** [review]

File: jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts (hunk 1 above)
Comment (Carminepo2) [reply jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts]: Yes, why not 😁

File: jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts (hunk 2 above)
Comment (Carminepo2) [reply jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts:78]: Yes, we are waiting for the `onboardedAt` information from selfcare

Comment (Carminepo2) [reply jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts:77]: Yes, we are waiting for the `onboardedAt` information from selfcare

Comment (Carminepo2) [reply jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts:78]: Thanks, I should probably add a comment

Comment (Carminepo2) [reply jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts:78]: UPDATE: Nevermind, I talked with @ruggerocastagnola, and he knows, everything is fine. Forget what I said about the `onboardedAt` property 😁

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**